gprofiler/containers_client.py

```diff
@@ -13,19 +14,18 @@
 logger = get_logger_adapter(__name__)
 
 
-class DockerClient:
+class ContainerNamesClient:
     def __init__(self) -> None:
         try:
-            self._client = docker.from_env()
-        except Exception:
+            self._containers_client = ContainersClient()
+        except NoContainerRuntimesError:
             logger.warning(
-                "Could not initiate the Docker client, so the profiling data will not include the container"
-                " names. If you are running gProfiler in a container, please mount the Docker sock file"
-                " by running the Docker run command with the following argument:"
-                ' "-v /var/run/docker.sock:/var/run/docker.sock". Otherwise, please open a new issue here:'
+                "Could not find a Docker daemon or CRI-compatible daemon, profiling data will not"
+                " include the container names. If you do have a containers runtime and it's not supported,"
+                " please open a new issue here:"
                 " https://github.com/Granulate/gprofiler/issues/new"
             )
-            self._client = None
+        logger.info(f"Discovered container runtimes: {self._containers_client.get_runtimes()}")
```

Comment:
Move this log into the `try` clause, it will fail if `ContainersClient()` raised an error.

Same for line 85. Did mypy not catch this? `self._containers_client` is `Optional[..]` now...

Reply:
Oh. I actually wanted it to print an empty list, But you're right - it won't work

Reply:
Mypy didn't catch it. Sadly it doesn't understand flows to this level, lol. Since `self._containers_client` is either set to non-None, or unset...